l1-contracts/contracts/bridge/L1SharedBridge.sol

```diff
@@ -134,124 +129,150 @@ contract L1SharedBridge is IL1SharedBridge, ReentrancyGuard, Ownable2StepUpgrade
-        if (_l1Token == ETH_TOKEN_ADDRESS) {
-            require(msg.value == _amount, "L1SharedBridge: msg.value not equal to amount");
-        } else {
-            // The Bridgehub also checks this, but we want to be sure
-            require(msg.value == 0, "ShB m.v > 0 b d.it");
-
-            uint256 amount = _depositFunds(_prevMsgSender, IERC20(_l1Token), _amount); // note if _prevMsgSender is this contract, this will return 0. This does not happen.
-            require(amount == _amount, "3T"); // The token has non-standard transfer logic
-        }
+        (address l1Asset, bytes32 assetInfo) = _getAssetProperties(_assetInfo);
+        _transferAllowanceToNTV(assetInfo, _amount, _prevMsgSender);
+        /* solhint-disable no-unused-vars */
+        bytes memory bridgeMintData = IL1StandardAsset(l1Asset).bridgeBurn{value: msg.value}({
+            _chainId: _chainId,
+            _mintValue: 0,
+            _assetInfo: assetInfo,
+            _prevMsgSender: _prevMsgSender,
+            _data: abi.encode(_amount, address(0))
+        });
 
-        if (!hyperbridgingEnabled[_chainId]) {
-            chainBalance[_chainId][_l1Token] += _amount;
-        }
         // Note that we don't save the deposited amount, as this is for the base token, which gets sent to the refundRecipient if the tx fails
-        emit BridgehubDepositBaseTokenInitiated(_chainId, _prevMsgSender, _l1Token, _amount);
+        emit BridgehubDepositBaseTokenInitiated(_chainId, _prevMsgSender, _assetInfo, _amount);
+        // We could use this data instead of the above
+        // emit BridgehubMintData(bridgeMintData);
+    }
+
+    /// @dev for backwards compatibility and to automatically register l1 assets, and we return the correct info.
+    function _getAssetProperties(bytes32 _assetInfo) internal returns (address l1Asset, bytes32 assetInfo) {
+        l1Asset = assetAddress[_assetInfo];
+        assetInfo = _assetInfo;
+        if (l1Asset == address(0) && (uint256(_assetInfo) <= type(uint160).max)) {
+            l1Asset = address(nativeTokenVault);
+            assetInfo = keccak256(abi.encode(block.chainid, NATIVE_TOKEN_VAULT_VIRTUAL_ADDRESS, _assetInfo));
+            nativeTokenVault.registerToken(address(uint160(uint256(_assetInfo))));
+        }
     }
 
-    /// @dev Transfers tokens from the depositor address to the smart contract address.
-    /// @return The difference between the contract balance before and after the transferring of funds.
-    function _depositFunds(address _from, IERC20 _token, uint256 _amount) internal returns (uint256) {
-        uint256 balanceBefore = _token.balanceOf(address(this));
-        // slither-disable-next-line arbitrary-send-erc20
-        _token.safeTransferFrom(_from, address(this), _amount);
-        uint256 balanceAfter = _token.balanceOf(address(this));
+    function decodeLegacyData(bytes calldata _data, address _prevMsgSender) external returns (bytes32, bytes memory) {
```

Comment:
also not to forget the access control, since it is not just a simple decoder, but it also moves allowances around